packages/opensrp-plans/src/components/ListView/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { OpenSRPService } from '../../helpers/dataLoaders';
+import { fetchPlanDefinitions, getPlanDefinitionsArray } from '../../ducks';
+import { connect } from 'react-redux';
+import { ColumnsType } from 'antd/lib/table/interface';
+import { PlansLoading, columns } from './utils';
+import { Link, RouteComponentProps } from 'react-router-dom';
+import { Store } from 'redux';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import { reducerName as PlansReducerName, reducer as PlansReducer } from '../../ducks';
+import { BrokenPage, useHandleBrokenPage } from '@opensrp/react-utils';
+import { Helmet } from 'react-helmet';
+import { PLANS_CREATE_VIEW_URL, RouteParams, TableColumnsNamespace } from '../../constants';
+import { CommonProps, defaultCommonProps } from '../../helpers/common';
+import { PlanDefinition } from 'opensrp-plans/src/plan-global-types';
+
+/** make sure product catalogue reducer is registered */
+reducerRegistry.register(PlansReducerName, PlansReducer);
+
+/** props for the PlansList view */
+interface Props<T = PlanDefinition> extends CommonProps {
+  data: T[];
+  productUnderView: T | null;
+  columns: ColumnsType<T>;
+  service: typeof OpenSRPService;
+  fetchPlansCreator: typeof fetchPlanDefinitions;
+}
+
+const defaultProps = {
+  ...defaultCommonProps,
+  data: [],
+  productUnderView: null,
+  columns: columns,
+  fetchPlansCreator: fetchPlanDefinitions,
+  service: OpenSRPService,
+};
+
+export type PlansListTypes = Props<PlanDefinition> & RouteComponentProps<RouteParams>;
+
+/** component that renders plans */
+
+const PlansList = (props: PlansListTypes) => {
+  const { service, data, columns, fetchPlansCreator, baseURL } = props;
+  const [loading, setLoading] = useState<boolean>(data.length === 0);
+  const { broken, errorMessage, handleBrokenPage } = useHandleBrokenPage();
+
+  useEffect(() => {
+    loadPlans(baseURL, service, fetchPlansCreator)
+      .catch((err: Error) => handleBrokenPage(err))
+      .finally(() => setLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (loading) {
+    return <PlansLoading />;
+  }
+
+  if (broken) {
+    return <BrokenPage errorMessage={errorMessage} />;
+  }
+
+  const pageTitle = `Active Missions`;
+  // add a key prop to the array data to be consumed by the table
+  const dataSource = data.map((singleObject) => {
+    const prodWithKey = {
+      ...singleObject,
+      key: `${TableColumnsNamespace}-${singleObject.date}`,
+    };
+    return prodWithKey;
+  });
+
+  return (
+    <div className="content-section">
+      <Helmet>
+        <title>{pageTitle}</title>
+      </Helmet>
+      <PageHeader title={pageTitle} className="page-header"></PageHeader>
+      <Row className={'list-view'}>
+        <Col className={'main-content'}>
+          <div className="main-content__header">
+            <Link to={PLANS_CREATE_VIEW_URL}>
+              <Button type="primary" size="large">
+                {' + New Mission'}
```

Comment:
How do you cater for the fact that while this is called "mission" for EUSM it is called plan everywhere else?

Reply:
@moshthepitt I think having an env variable that controls that will do. In this case, we can have it passed to the component as a prop from the client directory

Reply:
Please discuss that with the rest of the team so that we can have a collective decision on what to do about these.

Reply:
@moshthepitt  We had a discussion on this,  [multi-language support](https://docs.google.com/document/d/1WuOpq6WNwGNTrhBzYteW8BzcV8I6ZYgWXeOxqIXnKxE/edit) will handle it

Reply:
Cool, thanks.  Can you describe how, exactly, that document affects this piece of code in this PR?

Reply:
@moshthepitt @dubdabasoduba Since we will pull the different string literals from the lang file, for the different projects we need to establish a way in which lang files will be chosen based on the project. I do not have a concise solution to this. I think further scoping may be needed.

Reply:
> @moshthepitt @dubdabasoduba Since we will pull the different string literals from the lang file, for the different projects we need to establish a way in which lang files will be chosen based on the project. I do not have a concise solution to this. I think further scoping may be needed.

Looking into this. @moshthepitt I will set up a call with you to figure this out.

Reply:
In the meantime, Can we get this from a lang or constants module please?